### ISSUE TITLE
Permission error still persists on FAT32 drives

### ISSUE THREAD

**Issue by F483** [Storj/dataserv-client#117]
```
I:\storj_beta_B\dataserv_client_212\dataserv-client>dataserv-client --url=http:/
/status.driveshare.org:5000 --store_path I:\storj_beta_B\data --max_size=1999700000000  --use_folder_tree farm
INFO dataserv_client.messaging 42: Query: http://status.driveshare.org:5000/api/address
INFO dataserv_client.messaging 42: Query: http://status.driveshare.org:5000/api/register/1KtNaByCoA8RiYGaRbeGF1k2Vc369drYXr/1EcsAJ1LdZw3NyAzNWhgeYkh3ky2gqsAt5
WARNING dataserv_client.messaging 47: HTTPError()
INFO dataserv_client.api 171: Starting build
INFO dataserv_client.builder 101: Resuming from height 1
Traceback (most recent call last):
  File "dataserv-client", line 10, in <module>
  File "E:\dataserv-client\dataserv_client\cli.py", line 167, in main
  File "E:\dataserv-client\dataserv_client\api.py", line 213, in farm
  File "E:\dataserv-client\dataserv_client\api.py", line 194, in build
  File "E:\dataserv-client\dataserv_client\builder.py", line 121, in build
  File "E:\dataserv-client\dataserv_client\builder.py", line 69, in generate_shard
  File "E:\dataserv-client\dataserv_client\builder.py", line 54, in _get_shard_path
  File "E:\dataserv-client\dataserv_client\control\util.py", line 57, in ensure_path_exists
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 227, in makedirs
  File "C:\Python34\lib\os.py", line 237, in makedirs
PermissionError: [WinError 19] El medio está protegido contra escritura: 'I:\\st
orj_beta_B\\data\\33a'
```

**Comment by F483** [comment]
@littleskunk I thought this was solved on fat32 devices but *heunland* still gets a permission error. It looks like it fails creating the first folder of the tree `'I:\\st
orj_beta_B\\data\\33a'` and may be a different issue altogether.

**Comment by F483** [comment]
replicated the same error with a 64 GB brand new flash drive that is FAT32 formatted.

```I:\storj_beta_B\dataserv_client_212\dataserv-client>dataserv-client --url=http:/
/status.driveshare.org:5000 --store_path J:\data --max_size=10GB  --use_folder_t
ree farm
INFO dataserv_client.messaging 42: Query: http://status.driveshare.org:5000/api/
address
INFO dataserv_client.messaging 42: Query: http://status.driveshare.org:5000/api/
register/1KtNaByCoA8RiYGaRbeGF1k2Vc369drYXr/1EcsAJ1LdZw3NyAzNWhgeYkh3ky2gqsAt5
WARNING dataserv_client.messaging 47: HTTPError()
INFO dataserv_client.api 171: Starting build
INFO dataserv_client.builder 101: Resuming from height 1
Traceback (most recent call last):
 File "dataserv-client", line 10, in <module>
 File "E:\dataserv-client\dataserv_client\cli.py", line 167, in main
 File "E:\dataserv-client\dataserv_client\api.py", line 213, in farm
 File "E:\dataserv-client\dataserv_client\api.py", line 194, in build
 File "E:\dataserv-client\dataserv_client\builder.py", line 121, in build
 File "E:\dataserv-client\dataserv_client\builder.py", line 69, in generate_sha
rd
 File "E:\dataserv-client\dataserv_client\builder.py", line 54, in _get_shard_p
ath
 File "E:\dataserv-client\dataserv_client\control\util.py", line 57, in ensure_
path_exists
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 227, in makedirs
 File "C:\Python34\lib\os.py", line 237, in makedirs
PermissionError: [WinError 19] El medio está protegido contra escritura: 'J:\\da
ta\\33a'```

**Comment by F483** [comment]
Will retry on IOError exactly once, since it is sporadic. https://github.com/Storj/dataserv-client/commit/df760749f7bc5af9a6ed5d3c0a53a2dab9fafc15

**Comment by littleskunk** [comment]
At the moment i am at work. I will check it later.

My test was on a FAT32 RAM Drive. I did not create any folder for the test (--store_path=I:\). I used the good old build command. And last but not least i use the develop branch from git with a python 2.7 installation on windows. I did not use the final executeable build (with python 3.5 in it?)

**Comment by F483** [comment]
Sorry I commented on the wrong issue. Ignore my comment above.

**Comment by F483** [comment]
It appears that there is a difference in building with the client installed on a fat32 drive vs building to a fat32 drive.

**Comment by littleskunk** [comment]
Some new tests with FAT32 ram drive

without exception
--store_path I:\storj_beta_B\data 
farm command
python 2.7 and windows binary on the FAT32 drive

INFO dataserv_client.api 197: Current height at 14898.
INFO dataserv_client.api 208: Build finished

Next test will be with a slow USB Stick.

**Comment by littleskunk** [comment]
Full build without --use_folder_tree until permission denied cause of to many files. Start a new build with --use_folder_tree and you will get this exception.

WindowsError: [Error 82] Das Verzeichnis oder die Datei kann nicht erstellt werden: 'I:\\storj_beta_B\\data\\7b4'

With a CD Drive
WindowsError: [Error 5] Zugriff verweigert:

Not the same message but good for documentation.

I had no problems with a very old and very slow USB drive. Sry i dont know what else we sould try.

**Comment by super3** [comment]
Found this bit of information: 

The target file system is formatted as FAT32, however, I did look into that a little bit and didn't see how the limitations of that would have caused a problem. Naturally, none of my music files are over 4GB, and in total they do not exceed the 60,000 file limit per folder. My music collection is right around 20,000. So if how the drive is formatted is a problem, I'm not quite sure how.

Means that a FAT32 build will max out at 7TB in a folder.
